benchcheck_test.go

```diff
@@ -367,13 +362,129 @@ func TestStatBenchmarkResults(t *testing.T) {
 			got, err := benchcheck.Stat(tcase.oldres, tcase.newres)
 			assertNoError(t, err)
 
-			if diff := cmp.Diff(got, tcase.want, cmpfloats); diff != "" {
-				t.Fatal(diff)
+			assertEqualWithFloat(t, got, tcase.want)
+		})
+	}
+}
+
+func TestStatModule(t *testing.T) {
+	type rangef struct {
+		start float64
+		end   float64
+	}
+	type diff struct {
+		name  string
+		delta rangef
+	}
+	type result struct {
+		metric string
+		diffs  []diff
+	}
+	type testcase struct {
+		name   string
+		module string
+		oldver string
+		newver string
+		want   []result
+	}
+
+	t.Parallel()
+
+	if testing.Short() {
+		t.Skip("Skipping in short mode")
+		return
+	}
+
+	tcases := []testcase{
+		{
+			name:   "stat benchcheck",
+			module: "github.com/madlambda/benchcheck",
+			oldver: "0f9165271a00b54163d3fc4c73d52a13c3747a75",
+			newver: "e90da7b50cf0e191004809e415c64319465286d7",
+			want: []result{
+				{
+					metric: "time/op",
+					diffs: []diff{
+						{
+							name:  "Fake",
+							delta: rangef{start: -80, end: -70},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:   "stat benchcheck reversed versions",
+			module: "github.com/madlambda/benchcheck",
+			oldver: "e90da7b50cf0e191004809e415c64319465286d7",
+			newver: "0f9165271a00b54163d3fc4c73d52a13c3747a75",
+			want: []result{
+				{
+					metric: "time/op",
+					diffs: []diff{
+						{
+							name:  "Fake",
+							delta: rangef{start: 100, end: 400},
```

Comment:
This specific bench on macos on the CI env varies a LOT, maybe it is because we are running all tests in parallel..but disabling that makes things extremelly slower x_x. So in the end the ranges represents a ballpark =P. Outside the tests/not in macos in general the results are considerably precise from what I observed so far.

Reply:
Tried to remove parallelism and yet it oscilates a lot on mac/windows x_x. The env must be more limited for win/mac on GHA.